### ISSUE TITLE
Access blocked by CORS policy (Custom Jitsi server)

### ISSUE THREAD

**Issue by javieros105** [luvolondon/fvtt-module-jitsiwebrtc#62]
Hello. I've been having trouble with setting up my own jitsi server for foundry. I followed both the official Jitsi documentation and the guide by solfolango, trobuleshooted as much as possible but I don't know how to continue.
Just so you know Jitsi is configured with internal_hashed authentication for creating meetings.

The console shows the following 
<img width="913" alt="jitsiIssue" src="https://user-images.githubusercontent.com/62261473/105251925-2cff1c00-5b5b-11eb-8905-053cc1295d79.png">

Which is extremely weird because in between those CORS errors I do have some succesful binds
<img width="341" alt="jitsiIssue2" src="https://user-images.githubusercontent.com/62261473/105251929-31c3d000-5b5b-11eb-91dc-33da0205aa35.png">

From what I gather the module is able to create a room, if I use the bosh bind url it says it's succesful, but then it doesn't find it (?) when adding a user. So I'm very much lost

**Comment by bekriebel** [comment]
Make sure you have cross site access control (`Access-Control-Allow-*`) enabled for the BOSH `/http-bind` endpoint in your nginx configuration:

```
# BOSH
    location = /http-bind {
        proxy_pass      http://localhost:5280/http-bind;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header Host $http_host;
        add_header 'Access-Control-Allow-Origin' '*';
        add_header 'Access-Control-Allow-Credentials' 'true';
        add_header 'Access-Control-Allow-Methods' 'GET,HEAD,OPTIONS,POST,PUT';
        add_header 'Access-Control-Allow-Headers' 'Access-Control-Allow-Headers, Origin,Accept, X-Requested-With, Content-Type, Access-Control-Request-Method, Access-Control-Request-Headers';
    }
```

**Comment by javieros105** [comment]
I actually already did that and still no luck. 
![image](https://user-images.githubusercontent.com/62261473/105275986-2da9a980-5b7f-11eb-8920-e8025c5248fd.png)

But after putting the option for cross_domain_bosh in the cfg.lua to false it doesn't throw any cors error, but still not good.
![image](https://user-images.githubusercontent.com/62261473/105276154-7bbead00-5b7f-11eb-9141-cec4837522c9.png)
It gives me that error and gets stuck at asyc call to connect started

**Comment by javieros105** [comment]
So I reinstalled jitsi and using it without credentials it manages to get the conference going apparently, but I get (in all players) that they can see their own webcam, but nothing on the others. Jitsi manages to make the user join and set the video element but then it never adds the track to the element (using the public jitsi server it manages to connect perfectly well).

![image](https://user-images.githubusercontent.com/62261473/105387374-3ab4b000-5bf4-11eb-85e6-50789f5031a5.png)

**Comment by bekriebel** [comment]
What guide did you follow to set up your Jitsi server? It seems like there is something still misconfigured either on that server or on the server connection settings.

One additional thing to check is to make sure you do not have incorrect values set under the module settings. Unless there are unique values that aren't being set properly by your jitsi server's config, this option should be disabled:
![image](https://user-images.githubusercontent.com/183779/105408819-52843800-5be4-11eb-8cc0-2f6429e33f27.png).

If you are able to provide me with temporary credentials to your foundry instance and your jitsi server, I can try to take a look. To clarify, I don't want SSH credentials, but an FVTT gamemaster login and Jitsi credentials (unless you still have that set up with no authentication). I'm on the discord server as `bekit#4213`.

**Comment by javieros105** [comment]
So I actually managed to make it work without password. I forgot to do one thing to actually make me realize the mistake and that's trying the server in 2 different machines. Since I'm mounting it on a raspberry pi, I needed to recompile for arm, followed this guide https://gist.github.com/krithin/e50a6001c8435e46cb85f5c6c78e2d66 and that allowed me to make it work. Now when I continued with the original guide by solfolango it kinda broke it up for foundry so I'm gonna follow the official guide for authentication and see if it works after all that.

Thanks for all the help, I'm going to try that and report back. Maybe when I get a bit of time I'll compile a guide for mounting everything up on a RPI4

EDIT: Finally made it work, I had to change in the .lua file that conference had the option storage="memory" same as with the internal auth I think it was. Now it's working perfectly. Thanks so much for the help and also the awesome module!
